google/cloud/storage/object_stream.cc

```diff
@@ -61,25 +61,19 @@ ObjectWriteStream::ObjectWriteStream(
   init(buf_.get());
   // If buf_ is already closed, update internal state to represent
   // the fact that no more bytes can be uploaded to this object.
-  if (!buf_->IsOpen()) {
-    CloseBuf();
-  }
+  if (buf_ && !buf_->IsOpen()) CloseBuf();
```

Comment:
Just `if (IsOpen()) CloseBuf();`?

Reply:
If `IsOpen() == true` then `buf_ != nullptr` (that we want) and `buf_->IsOpen() == true` (that we do not want).  This line is supposed to handle the case where a stream is created with an already closed `buf_`. In that case we need to update the state, but if `buf_` is open then there is nothing to do.

I am not saying that the logic in this stuff is super clean, but I think it is correct.

Reply:
Oops ... sorry ... missed the `!`.